viper.go

```diff
@@ -789,6 +789,165 @@ func (v *Viper) GetSizeInBytes(key string) uint {
 	return parseSizeInBytes(sizeStr)
 }
 
+// Require returns the value associated with the given key or
+// an error if the key is not set.
+func Require(key string) (interface{}, error) { return v.Require(key) }
+func (v *Viper) Require(key string) (interface{}, error) {
+	value := v.Get(key)
+	if value == nil {
+		return nil, fmt.Errorf("key %s is not set", key)
```

Comment:
Since we are calling this "required" would it make more sense to just to panic in this circumstance?

Reply:
Panic is for errors from which the program cannot recover from. While in some cases this might be true for this Required function, when calling it somewhere in the middle of the program it might not be.

But TBH I don't think this function is a good idea. Viper is for consolidating and unmarshaling configuration. Validation shouldn't be a concern of Viper IMHO.

Reply:
While I do agree there are likely certain circumstances that would read a config in the middle of program execution, I think that's probably not the most common case to do so and then you would probably just not want use this functionality and use `GetX` instead.

I think for the vast majority of cases where this is useful, a program would have no way of recovering from a missing param being passed in (such as a missing DB_URL environment variable) and having this functionality can cause very useful error message during startup.